README.md

```diff
@@ -144,8 +171,8 @@ this file. If I need to do anything more that type a single command to run your
 code, include a script or makefile that will do the job.
 
 ## A2.5
-
-«replace this with your answer»
+Below is the command to run the file    
+python3 ./state.py
```

Comment:
The exercise asked you to express this table as code. I was looking for something like the code below, where the FSM is a data structure.

# -2


~~~ python
import unittest
from functools import reduce
ERROR  = "ERROR"
OK     = "OK"

states = {
  0:  { "the":    1 },
  1:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  2:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  3:  { "ate":    4, "ran":     4 },
  4:  { "slowly": 5, "noisily": 5, "EOI": OK },
  5:  { "EOI":    OK }
}

def getNextState(currentState, nextWord):
    return states.get(currentState,{}).get(nextWord, ERROR)

def stateMachine(sentence):
    return reduce(getNextState, sentence, 0) == OK

####################################################################################
class TestRunner(unittest.TestCase):
    def test_state(self):
        self.assertTrue(stateMachine(['the', 'smelly', 'cat', 'ate', 'slowly', 'EOI']))
        self.assertTrue(stateMachine(['the', 'lazy', 'smelly', 'dog', 'ate', 'noisily', 'EOI']))
        self.assertTrue(stateMachine(['the', 'dog', 'ate', 'EOI']))
        self.assertTrue(stateMachine(['the', 'smelly', 'cat', 'ate', 'EOI']))
        self.assertTrue(stateMachine(['the', 'lazy', 'smelly', 'lazy', 'dog', 'ate', 'EOI']))

        self.assertFalse(stateMachine(['the', 'cool', 'cat', 'ate', 'EOI']))
        self.assertFalse(stateMachine(['the', 'ate']))
        self.assertFalse(stateMachine(['the', 'dog', 'ate', 'slowly', 'noisily', 'EOI']))

unittest.main()
~~~